### DISCUSSION TITLE
Assertion for an element to be displayed

### DISCUSSION THREAD

**Discussion by brunohaller** [ui5-community/wdi5#566] [Q&A]
Hi,

with native webdriverIO selectors, we would usually use something like

`await expect(selector).toBeDisplayed()`
to wait for an element to appear and assert if it does not.

What is the closest wdi5 equivalent to this?

I assume that when using `asControl` it is already a blocking / failing operation and it would not make sense to execute `expect` on the result, at least not on the control itself (only text or other attributes after successfully finding a control).

Best Regards,
Bruno

**Comment by Siolto** [comment]
To assert that a control was really found with wdi5 you can do something like this:

```
const button = await browser.asControl(selector);
expect(button.isInitialized()).toBeTruthy()
```

**Reply by brunohaller** [reply]
What exactly is `isInitialized` checking? Is it checking whether a control isDisplayed?
